mlrun/api/api/endpoints/datastore_profile.py

```diff
@@ -82,8 +91,8 @@ async def list_datastore_profiles(
         project_name,
         auth_info.session,
     )
-    await mlrun.api.utils.auth.verifier.AuthVerifier().query_global_resource_permissions(
-        mlrun.common.schemas.AuthorizationResourceTypes.datastore_profile,
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_permissions(
+        project_name,
```

Comment:
Note that this check is not enough - it just verifies you have read access to the project. Usually what we do when listing objects would be to do this check first, and then after getting the list of objects, call `filter_project_resources_by_permissions` which keeps only objects that you have permissions on. Please do that here as well.